src/api-service/__app__/onefuzzlib/azure/containers.py

```diff
@@ -152,49 +152,58 @@ def delete_container(container: Container, storage_type: StorageType) -> bool:
 
 
 def get_container_sas_url_service(
-    container: Container,
-    service: BlockBlobService,
+    client: ContainerClient,
     *,
     read: bool = False,
-    add: bool = False,
-    create: bool = False,
     write: bool = False,
     delete: bool = False,
     list: bool = False,
+    delete_previous_version: bool = False,
+    tag: bool = False,
 ) -> str:
-    expiry = datetime.datetime.utcnow() + datetime.timedelta(days=30)
-    permission = ContainerPermissions(read, add, create, write, delete, list)
-
-    sas_token = service.generate_container_shared_access_signature(
-        container, permission=permission, expiry=expiry
+    account_name = client.account_name
+    container_name = client.container_name
+    account_key = get_storage_account_name_key_by_name(account_name)
+
+    sas = generate_container_sas(
+        account_name,
+        container_name,
+        account_key=account_key,
+        permission=ContainerSasPermissions(
+            read=read,
+            write=write,
+            delete=delete,
+            list=list,
+            delete_previous_version=delete_previous_version,
+            tag=tag,
+        ),
+        expiry=datetime.datetime.utcnow() + datetime.timedelta(days=30),
     )
 
-    url = service.make_container_url(container, sas_token=sas_token)
-    url = url.replace("?restype=container&", "?")
-    return str(url)
+    with_sas = ContainerClient(
+        f"https://{account_name}.blob.core.windows.net/",
```

Comment:
It may make sense to provide a constant for the default Azure blob service domain, or maybe a function that translates an account name to the default URL.

Reply:
added